Sources/ForageSDK/Foundation/Network/Collector.swift

```diff
@@ -400,19 +388,29 @@ class ForageVaultWrapper: VaultCollector {
                 request.setValue(header.value, forHTTPHeaderField: header.key)
             }
         }
-        request.httpBody = try! JSONSerialization.data(withJSONObject: body)
         
+        // measure the response time
         let measurement = VaultProxyResponseMonitor.newMeasurement(vault: VaultType.forage, action: vaultAction)
             .setPath(path)
             .setMethod(.post)
         
-        measurement.start()
-        
-        let task = URLSession.shared.dataTask(with: request) { data, response, error in
-            self.handleResponse(response: response, data: data, error: error, measurement: measurement, completion: completion)
+        // read the pin value from the main queue and add it to the request body
+        DispatchQueue.main.async {
+            guard let textElementValue = self.textElement.text else {
+                return completion(nil, CommonErrors.INCOMPLETE_PIN_ERROR)
+            }
+            body["pin"] = textElementValue
+            request.httpBody = try! JSONSerialization.data(withJSONObject: body)
+            
+            // go to a background queue for vault proxy request
+            DispatchQueue.global().async {
+                let task = URLSession.shared.dataTask(with: request) { data, response, error in
+                    self.handleResponse(response: response, data: data, error: error, measurement: measurement, completion: completion)
+                }
+                measurement.start()
+                task.resume()
+            }
         }
```

Comment:
Thank you for refactoring this @evanfreeze 🤩

Reply:
did a little more to get rid of the unneeded `DispatchQueue.global().async` stuff here: [f10e205](https://github.com/teamforage/forage-ios-sdk/pull/169/commits/f10e2054ff6ba5753ec3b463c1b8b61e235a577d)